spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -137,7 +138,19 @@
     }
     // Try catalog based name based resolution
     try {
-      return spark.table(tableName + "." + type);
+      if (tableName.startsWith("spark_catalog")) {
```

Comment:
Do we want to put this logic into a separate method like `loadMetadataUsingCatalog` or something?

Reply:
My only thought on why not to do this, is that hopefully in the future we get to remove the "startsWith(spark_catalog)" branch, and then the method is just spark.table(). But I can change it for now.